test-result-summary-client/src/TrafficLight/TrafficLight.jsx

```diff
@@ -50,6 +54,42 @@ function TrafficLight() {
             );
         }
     }
+
+    const buildJavaVersionMap = async (data, buildType, originBuildType) => {
+        let javaVersionMap = new Map();
+        for (const childData of data) {
+            childData.buildType = buildType;
+            const buildNameWithPlatform = childData.buildName.slice(
+                0,
+                lastIndexOf('_')
+            );
+            if (!javaVersionMap.has(buildNameWithPlatform)) {
+                let grandchildrenData = await fetchData(
+                    `/api/getChildBuilds?parentId=${childData._id}`
+                );
+                for (const grandchildData of grandchildrenData) {
+                    if (
+                        grandchildData.buildName.includes(
+                            `_${originBuildType}`
+                        ) &&
+                        grandchildData.javaVersion
+                    ) {
+                        const curbuildNameWithPlatform =
+                            removeAfterSecondLastUnderscore(
+                                grandchildData.buildName
+                            );
```

Comment:
I am not sure we need `removeAfterSecondLastUnderscore()`.  Instead of `curbuildNameWithPlatform`, use `buildNameWithPlatform`.

Reply:
Removed `removeAfterSecondLastUnderscore()` and used `buildNameWithPlatform`.